pkg/byteorder/byteorder.go

```diff
@@ -14,43 +14,7 @@
 
 package byteorder
 
-import (
-	"encoding/binary"
-	"fmt"
-	"reflect"
-)
-
-// reverse returns a reversed slice of b.
-func reverse(b []byte) []byte {
-	size := len(b)
-	c := make([]byte, size)
-
-	for i, j := size-1, 0; i >= 0; i, j = i-1, j+1 {
-		c[j] = b[i]
-	}
-
-	return c
-}
-
-// HostSliceToNetwork converts b to the networking byte order.
-func HostSliceToNetwork(b []byte, t reflect.Kind) interface{} {
-	switch t {
-	case reflect.Uint32:
-		if Native != binary.BigEndian {
-			return binary.BigEndian.Uint32(reverse(b))
-		}
-		return binary.BigEndian.Uint32(b)
-	case reflect.Uint16:
-		if Native != binary.BigEndian {
-			return binary.BigEndian.Uint16(reverse(b))
-		}
-		return binary.BigEndian.Uint16(b)
-	default:
-		panic(unsupported(t))
-	}
-}
-
-// unsupported returns a string to used for debugging unhandled types.
-func unsupported(field interface{}) string {
-	return fmt.Sprintf("unsupported type(%v): %v", reflect.TypeOf(field).Kind(), field)
+// HostSliceToNetwork32 converts b to the networking byte order.
+func HostSliceToNetwork32(b []byte) uint32 {
+	return Native.Uint32(b[len(b)-4:])
```

Comment:
why not `_ = b[3]` to assert the len of `b`?

Reply:
Good idea. And it turns out that this (and the original `HostSliceToNetwork` function) is completely wrong. Updated PR coming up...